core/src/main/java/hudson/model/AbstractBuild.java

```diff
@@ -742,13 +742,18 @@ protected final boolean performAllBuildSteps(BuildListener listener, Iterable<?
         }
 
         private void reportError(BuildStep bs, Throwable e, BuildListener listener, boolean phase) {
-            final String publisher = ((Publisher) bs).getDescriptor().getDisplayName();
+            final String buildStep;
+
+            if (bs instanceof Publisher)
+                buildStep = ((Publisher) bs).getDescriptor().getDisplayName();
```

Comment:
Is `getDescriptor` guaranteed to return something? If not you may have a NPE here.